diff --git a/cmake/Modules/merge_static_libs.cmake b/cmake/Modules/merge_static_libs.cmake
index 059f45b..92994f6 100644
--- a/cmake/Modules/merge_static_libs.cmake
+++ b/cmake/Modules/merge_static_libs.cmake
@@ -93,7 +93,29 @@ function(merge_static_libs target in_target)
             COMMENT "Merge static libraries with libtool"
             COMMAND xcrun libtool -static -o $<TARGET_FILE:${target}> -no_warning_for_no_symbols ${libs}
         )
-    elseif(CMAKE_C_COMPILER_ID MATCHES "^(Clang|GNU|Intel|IntelLLVM)$")
+    elseif(MSVC)
+        if(CMAKE_LIBTOOL)
+            set(BUNDLE_TOOL ${CMAKE_LIBTOOL})
+        else()
+            get_filename_component(COMPILER_DIRECTORY "${CMAKE_C_COMPILER}" DIRECTORY)
+            find_program(BUNDLE_TOOL llvm-lib HINTS "${COMPILER_DIRECTORY}" NO_DEFAULT_PATH)
+
+            if(NOT BUNDLE_TOOL)
+                find_program(BUNDLE_TOOL lib HINTS "${COMPILER_DIRECTORY}")
+            endif()
+
+            if(NOT BUNDLE_TOOL)
+                message(FATAL_ERROR "Cannot locate llvm-lib.exe or lib.exe to bundle libraries")
+            endif()
+        endif()
+
+        add_custom_command(
+            TARGET ${target}
+            POST_BUILD
+            COMMENT "Merge static libraries with lib.exe"
+            COMMAND ${BUNDLE_TOOL} /NOLOGO /OUT:$<TARGET_FILE:${target}> ${libs}
+        )
+    elseif(CMAKE_C_COMPILER_ID MATCHES "^(Clang|GNU|Intel|IntelLLVM)$")
         add_custom_command(
             TARGET ${target}
             POST_BUILD
@@ -114,22 +136,5 @@ function(merge_static_libs target in_target)
             COMMAND ${CMAKE_COMMAND} -E remove script.ar
         )
-    elseif(MSVC)
-        if(CMAKE_LIBTOOL)
-            set(BUNDLE_TOOL ${CMAKE_LIBTOOL})
-        else()
-            find_program(BUNDLE_TOOL lib HINTS "${CMAKE_C_COMPILER}/..")
-
-            if(NOT BUNDLE_TOOL)
-                message(FATAL_ERROR "Cannot locate lib.exe to bundle libraries")
-            endif()
-        endif()
-
-        add_custom_command(
-            TARGET ${target}
-            POST_BUILD
-            COMMENT "Merge static libraries with lib.exe"
-            COMMAND ${BUNDLE_TOOL} /NOLOGO /OUT:$<TARGET_FILE:${target}> ${libs}
-        )
     else()
         message(FATAL_ERROR "Unsupported platform for static link merging")
     endif()
